packaging/flavorgen/flavors/clusterclass_generators.go

```diff
@@ -304,6 +281,38 @@ func getClusterClassVariables() []clusterv1.ClusterClassVariable {
 			},
 		},
 	}
+
+	if !supervisorMode {
+		varForNoneSupervisorMode := []clusterv1.ClusterClassVariable{
```

Comment:
nit: we are already under if !supervisorMode
```suggestion
		variables := []clusterv1.ClusterClassVariable{
```

Reply:
Make sense to me, just rename it to `extraVariables` as we are already using variables :-)